### ISSUE TITLE
Globe rendering breaks after switching to mercator with Satellite Streets style. 

### ISSUE THREAD

**Issue by yunikkk** [mapbox/geojson.io#898]
Steps to reproduce : 
- pick `Mercator`, switch to `Globe`
- pick `Satellite streets` style, rendering breaks

Result : 

https://github.com/mapbox/geojson.io/assets/2265924/c5af143a-e2ea-420e-8442-c91cacf1a324

In the log there's an exception that looks relevant : 
```
mapbox-gl.js:33 Uncaught TypeError: Cannot read properties of undefined (reading 'get')
    at gE.continuePlacement (mapbox-gl.js:33:387682)
    at iS._updatePlacement (mapbox-gl.js:33:742518)
    at Map._render (mapbox-gl.js:33:844459)
    at mapbox-gl.js:33:851413
    at sentryWrapped (helpers.js:90:17)
```
Not sure this is the issue with geojson or gljs/style itself. 

Reproduced in Chrome 125.0.6422.176 / Safari 17.3.1 (19617.2.4.11.12).

**Comment by yunikkk** [comment]
Tracked in gljs internally.

**Comment by underoot** [comment]
It was already fixed in [GL JS v3.3.0](https://github.com/mapbox/mapbox-gl-js/releases/tag/v3.3.0). I updated GL JS and deployed, and it helped.
